packages/ra-ui-materialui/src/list/Pagination.tsx

```diff
@@ -77,7 +75,8 @@ const Pagination: FC<PaginationProps> = props => {
         [translate]
     );
 
-    if (total === 0) {
+    // Avoid rendering TablePagination if "page" value is invalid
+    if (total === 0 || page < 0 || page > totalPages) {
```

Comment:
Here, I think the test is `page < 1`, not `page < 0`
(The 0 based page number is in the handle function only)